contracts/aggregator/fantom/flashloan/implAaveV3/main.sol

```diff
@@ -0,0 +1,104 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+import "../helpers.sol";
+
+contract AaveImplementationFantom is Helper {
+
+    /**
+     * @dev Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @notice Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @param _tokens token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _route route for flashloan.
+     * @param _data extra data passed.
+    */
+    function flashLoan(	
+        address[] memory _tokens,	
+        uint256[] memory _amounts,
+        uint256 _route,
+        bytes calldata _data,
+        bytes calldata // kept for future use by instadapp. Currently not used anywhere.
+    ) external reentrancy {
+        require(_route == 9, "invalid-AAVE-route");
+        (_tokens, _amounts) = bubbleSort(_tokens, _amounts);
+        validateTokens(_tokens);
+        routeAaveV3(_tokens, _amounts, _data);
+    }
+
+    /**
+     * @dev Callback function for aave flashloan.
+     * @notice Callback function for aave flashloan.
+     * @param _assets list of asset addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets for flashloan.
+     * @param _premiums list of premiums/fees for the corresponding addresses for flashloan.
+     * @param _initiator initiator address for flashloan.
+     * @param _data extra data passed.
+    */
+    function executeOperation(
+        address[] memory _assets,
+        uint256[] memory _amounts,
+        uint256[] memory _premiums,
+        address _initiator,
+        bytes memory _data
+    ) external verifyDataHash(_data) returns (bool) {
+
+        require(_initiator == address(this), "not-same-sender");
+        require(msg.sender == aaveV3LendingAddr, "not-aave-sender");
+        FlashloanVariables memory instaLoanVariables_;
+
+        (uint256 route_, address sender_, bytes memory data_) = abi.decode(
+            _data,
+            (uint256, address, bytes)
+        );
```

Comment:
```suggestion
        (address sender_, bytes memory data_) = abi.decode(
            _data,
            (address, bytes)
        );
```
We dont need the route, do we?

Reply:
We have a single callback function for various routes so we use this variable there. But yes, in Fantom we presently only have a single route using Aave. I'll add this later on in case we add more routes.